### PR TITLE
Feat: AuthProvider model

### LINKED ISSUE #372: Model settings for SSO
## Background

We'll need to use a configuration object for a JavaScript library that helps implement the Login.gov flow. 

### Configuration data

Here's what the configuration object eventually needs to look like in JavaScript:

```js
// base address of Benefits application
const baseAddress = "https://localhost:8000";

// configuration object
var config = {
  // authorization server address
  authority: "https://login.gov",
  
  // this is registered application id with auth server
  client_id: "UUID",
  
  // post-login, user is redirected here
  redirect_uri: baseAddress + "/callback.html",
  
  // post-logout, user is redirected here
  post_logout_redirect_uri: baseAddress + "/index.html",
  
  // must be "code" to use the authorization-code flow
  response_type: "code",
  
  // space-delimited list
  // since you do not need any information (claims) about the user, "openid" is the only needed scope.
  // If you need refresh tokens, you must add the "offline_access" scope
  scope: "openid",
};
```

## Tasks

* [x] Build out the list of necessary config items (keys, URLs, endpoints, etc.) - *see above ⬆️*
* [x] Create a model like `AuthProvider` with text fields for the above list, all required
  * [ ] Consider storing URL parameters as the name of the app route, e.g. `eligibility:confirm` so we can use the Django helpers like `reverse()` and `{% url %}` and not duplicate raw paths in data 
* [x] Add a `ForeignKey` for `AuthProvider` to the `EligibilityVerifier` model, make it optional
* [x] Create a wrapper bool attribute on the `EligibilityVerifier` that denotes whether Auth is configured (the **"feature flag"**) by checking if there is a linked provider or not
* [x] Create attributes to track the Sign in + Sign out buttons  `msgid`s

### PR DESCRIPTION
**Pull request by angela-tran**
Closes #372

### CODE REVIEW

**Comment by thekaveman** [comment]
@angela-tran regarding the `silent*` fields - I think I mistakenly left these in from the sample. We probably won't need this feature (to silently renew in the background) so let's disregard these fields for now.

**Comment by thekaveman** [comment]
To your question about fixtures: is that maybe the reason the tests are failing? We should be able to start up the app with these changes, so if you need to add fixture(s) to make that work, go for it!

**Comment by thekaveman** [comment]
@angela-tran looking at this...

File: benefits/core/models.py [hunk 1]
```diff
@@ -95,6 +110,10 @@ def public_jwk(self):
         """jwcrypto.jwk.JWK instance of this Verifier's public key"""
         return self.public_key.jwk
 
+    @property
+    def requires_authentication(self):
+        return self.auth_provider.all().exists()
```
Comment (thekaveman) [inline benefits/core/models.py:115]: `self.auth_provider` is defined as a `ForeignKey`, not a `ManyToMany`

It should either have a value that is an instance of `AuthProvider`, or be `None`.

**Review by thekaveman — CHANGES_REQUESTED** [review]
Minor fix on the feature flag

**Review by thekaveman — CHANGES_REQUESTED** [review]
I added one more item to the checklist on #372: occurred to me that we'll need `msgid` for the Sign in and Sign out buttons.

Can you add attributes to `AuthProvider` for those please?

File: benefits/core/models.py (hunk 1 above)
Comment (angela-tran) [reply benefits/core/models.py:115]: Thanks for catching this. I was going off docs about resolving Django `QuerySet`s and didn't catch that `all()` doesn't make sense on `ForeignKey` fields 🤦 

Is the check for "is an instance of `AuthProvider`" necessary? I'm thinking we can just check that it is not `None`.

**Comment by angela-tran** [comment]
@thekaveman It is much easier to test code changes when we have an `AuthProvider` fixture so I'm thinking of creating a `0002_authproviders.json` file and bumping up the other fixture file numbers. I'm not sure what to use as values for the fixture though. Do you have any thoughts or advice?

File: benefits/core/models.py (hunk 1 above)
Comment (thekaveman) [reply benefits/core/models.py]: Yeah I think you're right, the `None` check is sufficient!

**Comment by thekaveman** [comment]
Yeah let's add the sample auth provider as a fixture here too 👍 

Sample values don't matter as much, you can take some of what was in the issue description, or make something up (as long as it is representative).

And since you're doing that, let's add a `name` attribute as well. We can use it in e.g. #377

**Comment by thekaveman** [comment]
@angela-tran if adding the sample fixture is a blocker here, we can do that in a follow-up. This PR is needed by a lot of other work so want to get it merged ASAP.

**Review by thekaveman — APPROVED** [review]

**Comment by angela-tran** [comment]
I added a bullet point to #375 to create the `AuthProvider` fixture.

**Pull request merged**